src/main/java/org/teamresistance/frc/Drive.java

```diff
@@ -0,0 +1,30 @@
+package org.teamresistance.frc;
+
+import org.strongback.components.ui.ContinuousRange;
+import org.strongback.drive.MecanumDrive;
+
+/**
+ * Created by shrey on 12/18/2016.
+ */
+public class Drive {
+
+  public final MecanumDrive robotDrive;
+
+  public Drive(MecanumDrive robotDrive) {
+    this.robotDrive = robotDrive;
+    States.driveState = DriveState.OP_CONTROL;
+  }
+
+  public void update(ContinuousRange translateXSpeed, ContinuousRange translateYSpeed, ContinuousRange rotation) {
+    if (States.driveState == DriveState.OP_CONTROL) {
+         this.robotDrive.cartesian(translateXSpeed.read(), translateYSpeed.read(), rotation.read());
+    } else if (States.driveState == DriveState.STOP) {
+      this.robotDrive.stop();
+    }
```

Comment:
When dealing with a state machine that should have a finite number of states, it is always appropriate to have at the end of the famous if/else or switch/case a line that says `throws new IllegalStateException()`, which this makes it easy to catch bugs caused by forgetting to cover a new state that was added to the enum.

```java
if (...) {
   ...
} else {
  throw new IllegalStateException();
}
```

or

```java
switch(...) {
  ...
  default:
    throw new IllegalStateException();
}
```